models/issue_dependency.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"time"
+)
+
+// IssueDependency is connection request for receiving issue notification.
+type IssueDependency struct {
+	ID           int64     `xorm:"pk autoincr"`
+	UserID       int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	IssueID      int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	DependencyID int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	Created      time.Time `xorm:"-"`
+	CreatedUnix  int64     `xorm:"INDEX created"`
+	Updated      time.Time `xorm:"-"`
+	UpdatedUnix  int64     `xorm:"updated"`
+}
+
+// Define Dependency Type Constants
+const (
+	DependencyTypeBlockedBy int64 = 1
+	DependencyTypeBlocking  int64 = 2
```

Comment:
But how do i check for the given type by the user in routers/repo/issue_dependency_remove.go:33?

Reply:
Please read all comments first. I think I've already answered it.

Reply:
Yes, but the compiler complains because i cannot compare an int with a custom type `DepenencyType` (although they both are ints).
Or did I overlooked something?

Reply:
You shouldn't have place where you compare int with custom type. What error?

Reply:
The error was something like "You cannot compare a variable of type `int` with a variable of type `DependencyType`". But it's ok i guess, i've solved it differently.